### PR TITLE
dropping MarkerIgnoringBase

### PR DESCRIPTION
**Pull request by dfa1**
@jamezp tentative implementation for marker support. Beware, there are lot of duplications right now! 

This PR can be merged once https://github.com/jboss-logging/jboss-logmanager/pull/335 is merged + released /cc @dmlloyd

### CODE REVIEW

**Comment by jamezp** [comment]
Thanks @dfa1. Overall I think this looks reasonable. I'd like to look at how the markers work for the `isXxxEnabled(Marker)` for my own information.

I think one way we can get this in before the log manager changes is using reflection for setting the marker. It's not ideal, but it would allow it to work with any version of the log manager.

**Comment by dfa1** [comment]
Tentative implementation for backward and forward compatibility using a static `MethodHandle`. Should also work when we will have `ExtLogRecord.setMarker` ... maybe I should add a unit test :-)

File: src/main/java/org/slf4j/impl/Slf4jLogger.java [hunk 1]
```diff
@@ -297,36 +491,109 @@ public void error(final String format, final Object... arguments) {
...
+
+    @Override
+    public void error(Marker marker, String msg, Throwable t) {
+        if (ALT_ERROR_INT < logger.getEffectiveLevel()) {
+            return;
+        }
+        log(null, org.jboss.logmanager.Level.ERROR, msg, t);
+    }
+
     protected Slf4jLogger readResolve() throws ObjectStreamException {
```
Comment (kieraCurtis) [inline src/main/java/org/slf4j/impl/Slf4jLogger.java:553]: I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Se: The readResolve method must be declared with a return type of Object. (SE_READ_RESOLVE_MUST_RETURN_OBJECT)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#se-the-readresolve-method-must-be-declared-with-a-return-type-of-object-se-read-resolve-must-return-object).
In order for the readResolve method to be recognized by the serialization mechanism, it must be declared to have a return type of Object.

Comment (dfa1) [reply src/main/java/org/slf4j/impl/Slf4jLogger.java:553]: good catch! thank you!

**Review by dmlloyd — APPROVED** [review]
It looks OK to me.

**Review by jamezp — APPROVED** [review]

**Comment by dfa1** [comment]
would be possible to merge to master and then release as beta? I will provide feedback for SLF4J at least.

**Comment by jamezp** [comment]
Sorry for the delay. I totally thought I merged this.

**Comment by jamezp** [comment]
@dfa1 I've release 1.2.0.Beta2. I messed up the 1.2.0.Beta1 release, so Beta2 is the way to go :) Hopefully this helps.

**Comment by dfa1** [comment]
thank you @jamezp, I will re-implement our special logger in the following weeks, by using https://github.com/quarkiverse/quarkus-logging-json and this new version of the logmanager.

Then I'll provide a similar PR for log4j, if you guys are still ok with that! /cc @dmlloyd

**Pull request merged**